tools/c7n_gcp/tests/conftest.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+import functools
+import os
+import shutil
+
+from c7n.testing import PyTestUtils, reset_session_cache, CustodianTestCore
+from c7n_gcp.client import Session, get_default_project, LOCAL_THREAD
+from recorder import sanitize_project_name, PROJECT_ID, HttpRecorder, HttpReplay
+from distutils.util import strtobool
+from pytest_terraform.tf import LazyPluginCacheDir, LazyReplay
+
+
+IS_C7N_FUNCTIONAL = strtobool(os.environ.get('C7N_FUNCTIONAL', 'no'))
+
+# If we have C7N_FUNCTIONAL make sure Replay is False otherwise enable Replay
+LazyReplay.value = not IS_C7N_FUNCTIONAL
+LazyPluginCacheDir.value = '../.tfcache'
+
+
+class GoogleFlightRecorder(CustodianTestCore):
```

Comment:
this base class feels like it belongs in tests/common.py or alternatively in c7n_gcp.testing

Reply:
I took some time to re-organize recorder.py, gcp_common.py and conftest.py to move the more re-usable components to `c7n_gcp.testing` to mimic how `c7n.testing` works

Reply:
yeah.. c7n.testing is intended to be generically useful to any provider, aws specific bits are non importable in tests/common.py so generally we want in tests/gcp_common.py to avoid paying distribution tax for serverless envs.

Reply:
This makes a lot of sense, I've removed the `c7n_gcp.testing` module and instead placed that all in `gcp_common` for now. If we need to move these to make them importable we can do that at that time